### PR TITLE
Update MultiNode with latest changes

### PR DESCRIPTION
**Pull request by DylanTinianov**
Updates MultiNode with the latest changes made during the Solana integration.
Ticket: https://smartcontract-it.atlassian.net/browse/BCFR-1069

New TransactionSender implementation uses generic RESULT and moves the error classification to the RPC client.

Also fixes the following issues:
1. Node lifecycle `OutOfSync` issue
2. Transaction Sender `SendTransaction` context issue
3. MultiNode DoAll context issue

### Soak Test
`CommitHash` ced19c25aa51cb6469248431e1f9f65e12f4abae (latest) 
--- |
Namespace: `soak-ocr-v2-arbitrum-sepolia-d3677 `
Num RPCs  Num RPCs  Latest Test
[Grafana Logs](https://grafana.ops.prod.cldev.sh/explore?schemaVersion=1&panes=%7B%22u3i%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22soak-ocr-v2-arbitrum-sepolia-d3677%5C%22%7D+%7C%3D++%60Node%60+or+%60TransactionSender%60+or+%60MultiNode%60+%7C+json+%7C+level%3D%7E%5C%22debug%7Cerror%7Ccrit%7Cwarn%5C%22%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22code%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-7d%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

File: common/client/node_lifecycle.go [hunk 1]
```diff
@@ -168,10 +168,8 @@ func (n *node[CHAIN_ID, HEAD, RPC]) aliveLoop() {
 				n.declareUnreachable()
 				return
 			}
-			_, latestChainInfo := n.StateAndLatest()
-			if outOfSync, liveNodes := n.isOutOfSyncWithPool(latestChainInfo); outOfSync {
+			if outOfSync, liveNodes := n.isOutOfSyncWithPool(); outOfSync {
```
Comment (patrick-dowell) [inline common/client/node_lifecycle.go:171]: Out of curiosity, why bundle the bug fix to the OutOfSync logic here with the Transaction Sender fixes below? They seem unrelated and could have been done as separate PRs. It also makes the individual PRs smaller and easier to review.

Comment (DylanTinianov) [reply common/client/node_lifecycle.go:171]: Fair question, in this case it was easier for me to copy all the changes over at once. I also needed to run an EVM soak test on all of these changes together. Triggering multiple soak tests for each part individually would be more complicated since there would be bugs without all of the changes in.

File: core/chains/evm/client/rpc_client.go [hunk 2]
```diff
@@ -802,7 +805,35 @@ func (r *RPCClient) BlockByNumberGeth(ctx context.Context, number *big.Int) (blo
...
+		txErr: err,
+	}
+	return result
+}
+
+func (r *SendTxResult) Error() error {
+	return r.err
+}
+
+func (r *SendTxResult) TxError() error {
```
Comment (dhaidashenko) [inline core/chains/evm/client/rpc_client.go]: Why do we need both TxError and Error?

File: common/client/transaction_sender_test.go [hunk 3]
```diff
@@ -17,37 +17,65 @@ import (
...
+	*MultiNode[types.ID, SendTxRPCClient[any, *sendTxResult]]
 }
 
 type sendTxRPC struct {
 	sendTxRun func(args mock.Arguments)
 	sendTxErr error
 }
 
-var _ SendTxRPCClient[any] = (*sendTxRPC)(nil)
+type sendTxResult struct {
```
Comment (dhaidashenko) [inline common/client/transaction_sender_test.go:31]: why do we need to define new sendTxResult. Can't we reuse existing one?

File: common/client/transaction_sender_test.go [hunk 4]
```diff
@@ -17,37 +17,65 @@ import (
...
-	return rpc.sendTxErr
+	return &sendTxResult{err: rpc.sendTxErr, txErr: rpc.sendTxErr, code: classifySendTxError(nil, rpc.sendTxErr)}
 }
 
 func newTestTransactionSender(t *testing.T, chainID types.ID, lggr logger.Logger,
-	nodes []Node[types.ID, SendTxRPCClient[any]],
-	sendOnlyNodes []SendOnlyNode[types.ID, SendTxRPCClient[any]],
-) (*sendTxMultiNode, *TransactionSender[any, types.ID, SendTxRPCClient[any]]) {
-	mn := sendTxMultiNode{NewMultiNode[types.ID, SendTxRPCClient[any]](
+	nodes []Node[types.ID, SendTxRPCClient[any, *sendTxResult]],
```
Comment (dhaidashenko) [inline common/client/transaction_sender_test.go]: consider creating a type alias for `SendTxRPCClient[any, *sendTxResult]` to make type defintion shorter

File: core/chains/evm/client/rpc_client.go [hunk 5]
```diff
@@ -122,7 +123,7 @@ type RPCClient struct {
 }
 
 var _ commonclient.RPCClient[*big.Int, *evmtypes.Head] = (*RPCClient)(nil)
-var _ commonclient.SendTxRPCClient[*types.Transaction] = (*RPCClient)(nil)
+var _ commonclient.SendTxRPCClient[*types.Transaction, *SendTxResult] = (*RPCClient)(nil)
 
 func NewRPCClient(
 	cfg config.NodePool,
```
Comment (dhaidashenko) [inline core/chains/evm/client/rpc_client.go:129]: we already pass NodePool config. We can avoid passing clientError as a separate param

File: common/client/transaction_sender_test.go [hunk 6]
```diff
@@ -357,11 +386,13 @@ func TestTransactionSender_SendTransaction_aggregateTxResults(t *testing.T) {
 		}
 
 		t.Run(testCase.Name, func(t *testing.T) {
-			_, txResult, err := aggregateTxResults(testCase.ResultsByCode)
-			if testCase.ExpectedTxResult == "" {
-				assert.NoError(t, err)
-			} else {
-				assert.EqualError(t, txResult, testCase.ExpectedTxResult)
+			txResult, err := aggregateTxResults(testCase.ResultsByCode)
+			if !testCase.ExpectedNilResult {
```
Comment (dhaidashenko) [inline common/client/transaction_sender_test.go]: Why do we need this field? Can't we just verify that txResult.TxError matches ExpectedTxResult, if `ExpectedTxResult` is not empty?

File: core/chains/evm/client/rpc_client.go (hunk 2 above)
Comment (DylanTinianov) [reply core/chains/evm/client/rpc_client.go]: Removed TxError

File: common/client/transaction_sender_test.go (hunk 3 above)
Comment (DylanTinianov) [reply common/client/transaction_sender_test.go:31]: Since this will be extracted to chainlink-framework, we shouldn't import the local EVM sendTxResult. This is also different from the Solana sendTxResult, which includes a signature field, so we don't want to move this to common. That's why I created the testing implementation separately.

**Review by jmank88 — APPROVED** [review]

**Review by dhaidashenko — APPROVED** [review]
